### DISCUSSION TITLE
How do you override the output directory in the command line?

### DISCUSSION THREAD

**Discussion by vurvdev** [premake/premake-core#1716] [Q&A] [answered]
Looking to get the contents of it through another program that will automatically invoke premake on unknown github repos.

**Comment by samsinsane** [comment, marked as answer]
Your question is pretty vague and could be interpreted many different ways.

- [Command line arguments](https://premake.github.io/docs/Command-Line-Arguments/)
- [`newoption`](https://premake.github.io/docs/newoption)
- [`location`](https://premake.github.io/docs/location/)?
- [`targetdir`](https://premake.github.io/docs/targetdir/)?

**Reply by vurvdev** [reply]
Wherever solution/project files are emitted, so I think ``location`` yeah, sorry about that, I don't use premake so I'm not familiar with this

**Reply by vurvdev** [reply]
This was seen as the answer but I'm looking for something directly I can modify in the command line, not a build script

I'm going to be running completely different scripts but expecting one output location, so I don't know how I could replicate ``location`` through that.

There doesn't seem to be a ``newoption`` to override this either

**Reply by samsinsane** [reply]
Read the Command line arguments link provided, it will show you how to use the `_OPTIONS` table as well as how to create your own command line options.

**Reply by vurvdev** [reply]
I'm not in control of what these premake scripts contain. So I have no idea if they will use _OPTIONS/newoption or not nor can I make my own options.

Trying to get the same output location for files like these which are completely different

https://github.com/danielga/gm_sourcenet/blob/master/premake5.lua
https://github.com/SuperiorServers/gm_tmysql4/blob/master/projects/premake5.lua
https://github.com/viral32111/gm_uuid/blob/master/scripts/premake5.lua

I can do this in cmake with ``-B\dir\here\`` for example

**Reply by samsinsane** [reply]
Premake doesn't currently support that. I'm not sure if there's a reason for that or not, @starkos might be able to shed some light on this? I would imagine that numerous APIs that contain paths relative to the output location would break - such as `postbuildcommands` and friends.

**Reply by starkos** [reply]
Yup, that's exactly why we leave support for relocatable targets up to the script authors. But you can always wrap the project's script in one of your own, and then apply your own settings, something like…

```lua
dofile('some_project/premake5.lua')

project('SomeProject')
   location 'some/other/location'
```

Which can then use command line options or whatever else you like to set the location. Does that help?

**Reply by vurvdev** [reply]
Yeah that sounds good! Thanks you both, sorry for my initial vague question
